tpm2_pytss/types.py

```diff
@@ -1546,15 +1547,34 @@ class TPM2B_PRIVATE_VENDOR_SPECIFIC(TPM2B_SIMPLE_OBJECT):
 
 
 class TPM2B_PUBLIC(TPM_OBJECT):
-    pass
+    @classmethod
+    def fromPEM(cls, data):
```

Comment:
I was thinking more like this, feel free to use `None` as a sentinel value to set defaults, but I want it to be clear in the builder whats getting done. Thoughts?
```python
fromPEM(cls, data, nameAlg=TPM2_ALG.SHA256, objectAttributes=(TPMA_OBJECT.DECRYPT | TPMA_OBJECT.SIGN_ENCRYPT | TPMA_OBJECT.USERWITHAUTH, scheme=None):
    ...
```

Reply:
that's better, the only issue is eccDetail.kdf, but looking at the comments in tss2_tpm_types.h it looks to be unused, just that it's required to be set to TPM2_ALG_NULL, so I'll keep the key for TPM2_ALG_ECC and just set it if true

Reply:
I'm good with this now, will merge and we can just do follow up patches. This moves the ball forward.